rust-runtime/aws-smithy-client/src/timeout.rs

```diff
@@ -51,8 +59,257 @@ impl Settings {
     /// Sets the read timeout
     pub fn with_read_timeout(self, read_timeout: Duration) -> Self {
         Self {
-            http_read_timeout: Some(read_timeout),
+            read_timeout: Some(read_timeout),
             ..self
         }
     }
 }
+
+#[derive(Clone, Debug)]
+/// A struct containing everything needed to create a new [`TimeoutService`]
+pub struct TimeoutServiceParams {
+    /// The duration of timeouts created from these params
+    duration: Duration,
+    /// The kind of timeouts created from these params
+    kind: &'static str,
+    /// The AsyncSleep impl that will be used to create time-limited futures
+    async_sleep: Arc<dyn AsyncSleep>,
+}
+
+#[derive(Clone, Debug, Default)]
+/// A struct of structs containing everything needed to create new [`TimeoutService`]s
+pub struct AllTimeoutServiceParams {
+    /// Params used to create a new API call [`TimeoutService`]
+    pub api_call: Option<TimeoutServiceParams>,
+    /// Params used to create a new API call attempt [`TimeoutService`]
+    pub api_call_attempt: Option<TimeoutServiceParams>,
+}
+
+/// Convert a [`TimeoutConfig`] into an [`AllTimeoutServiceParams`] in order to create the set of
+/// [`TimeoutService`]s needed by a [`crate::Client`]
+pub fn generate_timeout_service_params_from_timeout_config(
+    timeout_config: &TimeoutConfig,
```

Comment:
is there a way for a customer to provide an async sleep here?

Reply:
I've since added one but I don't see it in GH for some reason